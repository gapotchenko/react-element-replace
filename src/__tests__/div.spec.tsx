import * as React from 'react';
import { Replacer } from "..";
import { create } from 'react-test-renderer';


function TestComponent(props: {
    label?: string;
    children?: any;
}) {
    return (<div>
        {props.label} :{" "}
        <div style={{
            display: "inline-block",
            border: "1px solid black"
        }}>
            Label Me
      </div>
        <div>{props.children}</div>
    </div>);
}

test("div color replace", () => {
    const component = create(
        <Replacer
            matchElement={"div"}
            replace={({ type: Div, props }: any) => (
                <Div
                    {...props}
                    style={{
                        ...props.style,
                        color: "blue",
                        padding: "3px"
                    }}
                />
            )}
        >
            <TestComponent>
                <TestComponent />
            </TestComponent>
            <TestComponent />
        </Replacer>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
