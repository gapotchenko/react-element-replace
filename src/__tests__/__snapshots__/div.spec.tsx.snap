// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`div color replace 1`] = `
Array [
  <div
    style={
      Object {
        "color": "blue",
        "padding": "3px",
      }
    }
  >
     :
     
    <div
      style={
        Object {
          "border": "1px solid black",
          "color": "blue",
          "display": "inline-block",
          "padding": "3px",
        }
      }
    >
      Label Me
    </div>
    <div
      style={
        Object {
          "color": "blue",
          "padding": "3px",
        }
      }
    >
      <div
        style={
          Object {
            "color": "blue",
            "padding": "3px",
          }
        }
      >
         :
         
        <div
          style={
            Object {
              "border": "1px solid black",
              "color": "blue",
              "display": "inline-block",
              "padding": "3px",
            }
          }
        >
          Label Me
        </div>
        <div
          style={
            Object {
              "color": "blue",
              "padding": "3px",
            }
          }
        />
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "color": "blue",
        "padding": "3px",
      }
    }
  >
     :
     
    <div
      style={
        Object {
          "border": "1px solid black",
          "color": "blue",
          "display": "inline-block",
          "padding": "3px",
        }
      }
    >
      Label Me
    </div>
    <div
      style={
        Object {
          "color": "blue",
          "padding": "3px",
        }
      }
    />
  </div>,
]
`;
